### PR TITLE
Restore compatibility with GHC-7.8

### LINKED ISSUE #41: Build failures with GHC-7.8
I've sadly lost my build logs but when I tried to build with `lts-2.22` I ran into a bad import from `math-functions` and missing imports of `pure`.

Currently the cabal file claims that you're testing with GHC-7.8 so you should do probably that. :)

Nice package BTW. Looks very useful for benchmarking dependencies of `criterion`! :)

### PR DESCRIPTION
**Pull request by sjakobi**
Fixes #41.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
@sjakobi Thanks for the PR. Looks good to me, except that we can put the new imports under a preprocessor conditional to avoid warnings due to `-Wunused-imports` (in appveyor build). 

I guess we should use the same compiler options across travis builds as well, I will try to fix that by using the same script (`package-test`) for travis that we use for appveyor.

File: Gauge/Source/RUsage.hsc [hunk 1]
```diff
@@ -147,10 +148,8 @@ clongToW64 = fromIntegral
 sizeRUsage :: Int
 sizeRUsage = #const sizeof(struct rusage)
 
-pattern Self :: Who
```
Comment (vincenthz) [inline Gauge/Source/RUsage.hsc]: This one looks slightly dangerous. Is that necessary for 7.8, or is that the syntax is different on 7.8 so it's easier to remove the signature than doing a conditional signature ?

**Comment by vincenthz** [comment]
I'm slightly on the fence here, 7.8 is technically out of the common 3 compilers support already ..
Maybe by refactoring the imports to a common import file would prevent the issue of unused imports *and* be a useful cleanup going forward

**Comment by vincenthz** [comment]
Ok, maybe this one make sense afterall; Since a benchmarking tool is commonly used to benchmark multiple ghc versions, it's sensible to keep supporting 7.8.

File: Gauge/Source/RUsage.hsc (hunk 1 above)
Comment (sjakobi) [reply Gauge/Source/RUsage.hsc]: I'm afraid I can't find a reference on the syntax change but with GHC-7.8 I get

```
/home/simon/src/hs-gauge/Gauge/Source/RUsage.hsc:151:14: parse error on input ‘::’
```

How about

```
pattern Self = (#const RUSAGE_SELF) :: Who
```

for a compromise? 

I can also include the signature conditionally with CPP.

**Comment by sjakobi** [comment]
> Looks good to me, except that we can put the new imports under a preprocessor conditional to avoid warnings due to -Wunused-imports (in appveyor build).

@harendra-kumar How about I apply [the "import-Prelude-last-hack"](https://ghc.haskell.org/trac/ghc/wiki/Migration/7.10#GHCsaysTheimportof...isredundant)? That would save some CPP.

File: Gauge/Source/RUsage.hsc (hunk 1 above)
Comment (phadej) [reply Gauge/Source/RUsage.hsc]: https://ghc.haskell.org/trac/ghc/ticket/8584 went in GHC-7.10

**Review by phadej — COMMENTED** [review]
Could you also bump the lower bound to `base >= 4.7`, because it seems there aren't install plans with GHC-7.6, and they aren't tested.

**Comment by sjakobi** [comment]
Thanks for the pointer @phadej! :+1: I have adjusted the lower bound for `base`.

**Comment by sjakobi** [comment]
I don't understand why AppVeyor is reporting missing signatures for the pattern synonyms again.

**Comment by vincenthz** [comment]
@sjakobi the pattern synonyms are defined twice in this file. one when SUPPORT_RUSAGE is defined, and once when it's not (i.e. windows). the later is the one that remain to be tweaked

**Comment by sjakobi** [comment]
Thanks @vincenthz! :)

**Comment by harendra-kumar** [comment]
@sjakobi yes, the `import Prelude` trick is cleaner. The change looks good to go.

We can also put a `tested-with` field in the .cabal file to explicitly indicate the ghc versions we have tested it with.

**Comment by sjakobi** [comment]
> We can also put a tested-with field in the .cabal file to explicitly indicate the ghc versions we have tested it with.

It's already in there and up to date.

**Comment by vincenthz** [comment]
thanks for the PR

**Comment by sjakobi** [comment]
Thanks for taking on the maintenance burden for another GHC version! :+1:

**Comment by vincenthz** [comment]
ah ! here I thought you were volunteering to help maintain ;)

**Comment by sjakobi** [comment]
Hm, well I won't promise anything but I have starting watching the repo. :)

**Pull request merged**